### ISSUE TITLE
Using different e-link templates and sparql endpoints

### ISSUE THREAD

**Issue by fsasaki** [freme-project/CKEditor-Plugin#24]
Is it possible to use a different e-link template or even to let the user choose a template? Looking at plugin.js 
it seems that "all: 4477" is hard coded. My use case is to have data from a non dbpedia endpoint (e.g. JRC names) and potentially several endpoints in parallel, let the user apply several templates at the same time.

**Comment by bjdmeest** [comment]
I would suggest adding this to the e-link dialog.
Instead of having one tab with one template at the moment, have a list of
tabs, and make the templates configurable per tab in the plugin code.

So, you would have something like

```
link: {
    templates: [
        {
            id: 4477,
            title: 'DBpedia'
            properties: [
                {
                    types: ['http://nerd.eurecom.fr/ontology#Person', '
http://dbpedia.org/ontology/Person'],
                    data: [
                        ["abstract", "http://dbpedia.org/ontology/abstract
"],
                        ["date of birth", "
http://dbpedia.org/ontology/birthDate"],
                        [...]
                    ]
                },
                {
                    types: ['http://nerd.eurecom.fr/ontology#Location', '
http://dbpedia.org/ontology/PopulatedPlace', '
http://dbpedia.org/ontology/Location'],
                    data: [
                        ["abstract", "http://dbpedia.org/ontology/abstract
"],
                        ["thumbnail", "http://dbpedia.org/ontology/thumbnail
"],
                        [...]
                    ]
                }
            ]
        },
        {
            id: 1234,
            title: 'JRC'
            properties: [...]
        }
    ],
}
```

and the e-link dialog would then have two tabs, one for dbpedia and one for
JRC. The user can browse tabs and insert the information of the templates
he/she choses.

Thoughts?

2016-09-15 8:16 GMT+02:00 Pieter Heyvaert <notifications@github.com>:

**Comment by fsasaki** [comment]
+1

**Comment by fsasaki** [comment]
@bjdmeest , would it be possible to have this ready for the Frankfurt book fair?

**Comment by bjdmeest** [comment]
That should be possible, expect changes by the end of next week (holidays
atm)! :)

Op 25 sep. 2016 4:18 p.m. schreef "Felix Sasaki" <notifications@github.com>:

**Comment by fsasaki** [comment]
Great, thanks :)

**Comment by bjdmeest** [comment]
Did this in the dev branch (d0505ebaebdf290024c60bd8d74325fc3c83a8b9). Maybe you could play with the configuration and see whether this fulfills the request? If so, I'd be happy to merge it into master as well.
Live on https://api-dev.freme-project.eu/ckeditor-dev/ckeditor/samples/freme.html (but you don't see any changes because there's only one template configured atm :) )

**Comment by fsasaki** [comment]
Looks good to me, thanks! When you merge it into the master, could you add a template, so that people see the functionality?

**Comment by bjdmeest** [comment]
Sure, any suggestions for a second template? :)

**Comment by fsasaki** [comment]
How about the template with ID 2? See
curl -X GET --header 'Accept: application/json' '
https://api-dev.freme-project.eu/current/e-link/templates/2'

2016-10-06 14:34 GMT+02:00 Ben De Meester <notifications@github.com>:

**Comment by bjdmeest** [comment]
Template 2 didn't seem to work immediately, but I added a commit (3568a75cc4551aeb430ad7e9a638ddeb2fcadb1c) with a second template, if everything is fine, I'll merge

**Comment by fsasaki** [comment]
Great, thanks. Feel free to merge.

2016-10-06 16:14 GMT+02:00 Ben De Meester <notifications@github.com>:

**Comment by bjdmeest** [comment]
Fixed by ffdbf74677a250927b5b45203c946a494fe94bd4
